src/test/java/io/jenkins/security/CompatibilityTest.java

```diff
@@ -1,109 +1,53 @@
 package io.jenkins.security;
 
 import java.io.ByteArrayOutputStream;
+import java.io.File;
 import java.io.IOException;
 import java.io.InputStream;
 import java.net.URL;
+import java.util.Arrays;
+import java.util.Objects;
 import java.util.stream.Stream;
 import java.util.zip.ZipEntry;
+import java.util.zip.ZipFile;
 import java.util.zip.ZipInputStream;
 import org.apache.commons.io.IOUtils;
 import org.junit.jupiter.params.ParameterizedTest;
 import org.junit.jupiter.params.provider.Arguments;
 import org.junit.jupiter.params.provider.MethodSource;
 
 import static org.junit.jupiter.api.Assertions.assertNotNull;
+import static org.junit.jupiter.api.Assertions.assertTrue;
 import static org.junit.jupiter.api.Assertions.fail;
 
 public class CompatibilityTest {
+
+    private static final String REMOTING_JAR_NAME_PATTERN = "remoting[-](.+)[.]jar";
+    private static final String REMOTING_JAR_DIRECTORY = "target/classes/old-remoting/";
```

Comment:
Note that it is necessary to hard-code this; you can look up _one_ known file (e.g., `remoting-4.0.jar`) using `getResource`, convert that `URL` to `URI` then `File` or `Path`, then search the parent dir.

Reply:
Good idea!